baseplate/experiments/__init__.py

```diff
@@ -134,25 +143,28 @@ def variant(self, name, user=None, bucketing_event_override=None,
         :rtype: :py:class:`str`
         :return: Variant name if a variant is active, None otherwise.
         """
+
+        inputs = dict(kwargs)
         experiment = self._get_experiment(name)
 
         if experiment is None:
             return None
 
-        inputs = dict(kwargs)
```

Comment:
I would leave this here, if the experiment doesn't exist, then you can skip copying the dict.

Reply:
fixed. Didn't realize it's a deep copy and seemed cleaner up above.

Reply:
Slight correction, it's not _technically_ a deep copy since it doesn't recursively other containers within the dict:

```python
In [1]: x = {1: 'test', 2: ['a', 'b']}

In [2]: y = dict(x)

In [3]: x is y
Out[3]: False

In [4]: y[2] is x[2]
Out[4]: True

In [5]: from copy import deepcopy

In [6]: z = deepcopy(x)

In [7]: z[2] is x[2]
Out[7]: False

In [16]: y[2].append('c')

In [17]: y
Out[17]: {1: 'test', 2: ['a', 'b', 'c']}

In [18]: x
Out[18]: {1: 'test', 2: ['a', 'b', 'c']}

In [19]: z
Out[19]: {1: 'test', 2: ['a', 'b']}
```